docs/source/release_notes/release-notes.rst

```diff
@@ -44,6 +44,25 @@ develop
     *    - Type
          - Change
 
+    *    - |improved|
+         - Timelock server now proceses lock requests using async Jetty servlets, rather than blocking request threads. This leads to more stability and higher throughput during periods of heavy lock contention.
+           Timelock server config has a new ``useAsyncLockService`` option that can be used to switch between the new and old lock service implementation. This option defaults to ``true``.    
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/2084>`__)
+           
+    *    - |devbreak|
```

Comment:
Nit: This would probably also be 'improved', 'now configurable' usually sounds like a good thing!